src/controller/CHIPDeviceController.cpp

```diff
@@ -880,6 +880,20 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, RendezvousParam
     VerifyOrExit(mDeviceBeingPaired < kNumMaxActiveDevices, err = CHIP_ERROR_NO_MEMORY);
     device = &mActiveDevices[mDeviceBeingPaired];
 
+    // If the CSRNonce is passed in, using that else using a random one..
+    if (params.HasCSRNonce())
+    {
+        device->SetCSRNonce(params.GetCSRNonce());
+        VerifyOrReturnError(device->SetCSRNonce(params.GetCSRNonce()), CHIP_ERROR_INVALID_ARGUMENT);
+    }
+    else
+    {
+        uint8_t mCSRNonce[kOpCSRNonceLength];
+        Crypto::DRBG_get_bytes(mCSRNonce, sizeof(mCSRNonce));
+        chip::Optional<chip::ByteSpan> opCSROptional(ByteSpan(mCSRNonce, sizeof(mCSRNonce)));
```

Comment:
Ignoring the Optional bits for now, just `ByteSpan(mCSRNonce)` will do the right thing in terms of picking up the size in this case and is safer (in that it ensures the size can't get out of sync, since it determines it from the type of `mCSRNonce`).

Reply:
Done